src/state_manager/mod.rs

```diff
@@ -526,23 +526,53 @@ where
                     tipset.epoch(),
                     tipset.len(),
                 );
+
+                // First, try to look up the state and receipt if not found in the blockstore
+                // compute it
+                if let Some(state_from_child) =
+                    self.try_lookup_state_from_next_tipset(tipset.as_ref())
+                {
+                    return Ok(state_from_child);
+                }
+
+                trace!("Computing state for tipset at epoch {}", tipset.epoch());
                 let state_output = self
                     .compute_tipset_state(Arc::clone(tipset), NO_CALLBACK, VMTrace::NotTraced)
                     .await?;
                 for events_root in state_output.events_roots.iter().flatten() {
                     trace!("Indexing events root @{}: {}", tipset.epoch(), events_root);
-
                     self.chain_store().put_index(events_root, key)?;
                 }
+
+                self.update_receipt_and_events_cache(key, &state_output);
```

Comment:
Do we really need to do this? That means even if we disable the RPC server, we're still going to fill some receipt and events cache, which could get quite large. What do you think?

Reply:
Nice catch.

You're correct, cache won't be required if the RPC is disabled.

I guess we can remove it from this PR, we can tackle this separately.

Just putting a thought here:
 Could we do something like initalize cache only if `RPC` is enabled, and if the cache is initalized then only insert the data into it?. @elmattic what do you think?.

Reply:
That said, we could issue a warning upon exit if the cache is not empty and RPC is disabled.

Reply:
Lets do one thing, I will remove the `updating cache` part from this PR so we can track it separately.

Reply:
Raised an Issue for this: https://github.com/ChainSafe/forest/issues/5626